nntrainer/src/addition_layer.cpp

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0-only
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	addition_layer.cpp
+ * @date	30 July 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Addition Layer Class for Neural Network
+ *
+ */
+
+#include <addition_layer.h>
+#include <layer.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+int AdditionLayer::initialize(bool last) {
+  int status = ML_ERROR_NONE;
```

Comment:
Fail when num_input is not set?

Reply:
As you already know, num_inputs is initialized with value of 1.
So, there is no way to know if it is uninitialized. One might argue that if `num_inputs` is 1, this layer can fail, but rather, it should be optimized out later by the graph.

Reply:
How about set default value of num_inputs to be 0? Most of the time, setting sensible default works for other properties. However, for this case, I don't see the use case of num_inputs with default value are used. If there is no use case for default value, num_input must be set before used properly and we should validate accordingly as well.

Reply:
done.